inference-engine/thirdparty/clDNN/tests/test_cases/gemm_gpu_test.cpp

```diff
@@ -3232,8 +3232,162 @@ TEST(gemm_gpu, basic_smarcink2) {
+
+    for (auto b = 0; b < b_out_num; ++b) {
+        for (auto f = 0; f < f_out_num; ++f) {
+            for (auto i = 0; i < m_size; ++i) {
+                for (auto j = 0; j < n_size; ++j) {
+                    int input2_data_index = getGemmIndex(j, i, f, b, x2_size, y2_size, f2_num, b2_num, x2_pitch, y2_pitch, f2_pitch, b2_pitch);
+                    int out_data_index = getGemmIndex(j, i, f, b, x_out_size, y_out_size, f_out_num, b_out_num,
+                                                      x_out_pitch, y_out_pitch, f_out_pitch, b_out_pitch);
+                    out_data[out_data_index] = 0;
+
+                    for (auto k = 0; k < k_size; ++k) {
+                        int input0_data_index = getGemmIndex(k * (!transpose_input0) + i * transpose_input0, i * (!transpose_input0) +
+                        k * transpose_input0, f, b, x0_size, y0_size, f0_num, b0_num, x0_pitch, y0_pitch, f0_pitch, b0_pitch);
+                        int input1_data_index = getGemmIndex(j * (!transpose_input1) + k * transpose_input1, k * (!transpose_input1) +
+                        j * transpose_input1, f, b, x1_size, y1_size, f1_num, b1_num, x1_pitch, y1_pitch, f1_pitch, b1_pitch);
+
+                        out_data[out_data_index] += (float)(input0_data_bfyx[input0_data_index]) * (float)(input1_data_bfyx[input1_data_index]);
+                    }
+
+                    out_data[out_data_index] *= alpha;
+                    out_data[out_data_index] += beta * (float)(input2_data_bfyx[input2_data_index]);
+                }
+            }
+        }
+    }
+
+    topology topology;
+    topology.add(
+        input_layout("input0", input0_mem.get_layout())
+    );
+    topology.add(
+        input_layout("input1", input1_mem.get_layout())
+    );
+    topology.add(
+        input_layout("input2", input2_mem.get_layout())
+    );
+    topology.add(
+        gemm("output", { "input0", "input1", "input2" }, data_types::f32, transpose_input0, transpose_input1, alpha, beta)
+    );
+
+    build_options options;
+    implementation_desc gemm_int8_impl = { format::bfyx, "gemm_mmad_int8" };
+    options.set_option(build_option::force_implementations({ {"output", gemm_int8_impl} }));
+
+    network network(engine, topology, options);
+    network.set_input_data("input0", input0_mem);
+    network.set_input_data("input1", input1_mem);
+    network.set_input_data("input2", input2_mem);
+    auto outputs = network.execute();
+
+    auto output = outputs.at("output").get_memory();
+    auto output_ptr = output.pointer<float>();
+
+    EXPECT_EQ(output_ptr.size(), (uint32_t)(b_out_num * f_out_num * m_size * n_size));
+    for (uint32_t i = 0; i < out_data.size(); ++i) {
         EXPECT_FLOAT_EQ(output_ptr[i], out_data[i]);
     }
 }
 
+// Gemm int8 transposing tests
+TEST(gemm_int8_transpose_tests, nn_case) { generic_gemm_int8_test(64, 64, 64, 1, 2, 1, 2, 1, 2, 1, 2, false, false, 1.5f, 2.0f); }
```

Comment:
Please use parameterized tests for such cases (`TEST_P` + `INSTANTIATE_TEST_CASE_P`)